de.metas.adempiere.adempiere/base/src/test/java/de/metas/location/geocoding/asynchandler/LocationGeocodeEventHandlerTest.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * #%L
+ * de.metas.adempiere.adempiere.base
+ * %%
+ * Copyright (C) 2019 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.location.geocoding.asynchandler;
+
+import de.metas.event.IEventBusFactory;
+import de.metas.location.LocationId;
+import de.metas.location.geocoding.GeoCoordinatesRequest;
+import de.metas.location.geocoding.GeocodingService;
+import de.metas.location.geocoding.GeographicalCoordinates;
+import mockit.Expectations;
+import mockit.Injectable;
+import mockit.Tested;
+import org.adempiere.model.InterfaceWrapperHelper;
+import org.adempiere.test.AdempiereTestHelper;
+import org.compiere.model.I_C_Country;
+import org.compiere.model.I_C_Location;
+import org.compiere.model.X_C_Location;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.math.BigDecimal;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Must use jUnit 4 because our version of junit5 and jmockit are incompatible and an error is thrown.
```

Comment:
just as a sidenote: when i mock nowadaws, i tend to use mockito; it's better integrates with spring (and afair junit5).

Reply:
will update the test when i get back to this, during the integration with frontend.